src/modelfree/scheduling.py

```diff
@@ -1,31 +1,55 @@
 from abc import ABC, abstractmethod
+import collections
 import functools
+import itertools
+import operator
+
+
+def _validate_func_type(func_type):
+    allowed_func_types = frozenset(['lr', 'rew_shape', 'noise'])
+    if func_type not in allowed_func_types:
+        raise KeyError("func_type not in allowed_func_types")
 
 
 class Scheduler(object):
     """Keep track of frac_remaining and return time-dependent values"""
-    def __init__(self, func_dict=None):
-        if func_dict is None:
-            func_dict = {}
-        assert isinstance(func_dict, dict)
-        self.func_dict = func_dict
-        self.allowed_func_types = ['lr', 'rew_shape', 'noise']
+    def __init__(self, annealer_dict=None):
+        if annealer_dict is None:
+            annealer_dict = {}
+        self.annealer_dict = annealer_dict
+        self.func_dict = {}
```

Comment:
Do we actually need `func_dict`? Can't we just have the functions inside of here call `get_value` on the annealer directly?